src/4/Maybe.test.ts

```diff
@@ -45,6 +45,46 @@ describe(Maybe, () => {
           .map((p) => p * Maybe.some(taxRate).valueOr(1))
           .valueOr(0);
 
+    class ApplyTaxesByRate {
+      constructor(public readonly value: number) {}
+
+      bind(callback: (n: number) => ApplyTaxesByRate) {
+        const newValue = callback(this.value).value;
+        return new ApplyTaxesByRate(newValue);
+      }
+
+      isNone() {
+        return this.value === null || this.value === undefined;
+      }
+
+      isSome() {
+        return !this.isNone();
+      }
+
+      map(callback: (n: number) => number) {
+        return this.isSome()
+          ? new ApplyTaxesByRate(callback(this.value))
+          : this;
+      }
+
+      valueOrZero() {
+        return this.isSome() ? this.value : 0;
+      }
+
+      static from(n: number) {
+        return new ApplyTaxesByRate(n);
+      }
+    }
+
+    const foo = (taxRate: number) => (price: number) =>
+      ApplyTaxesByRate.from(taxRate)
+        .map((n) => n * price)
+        .valueOrZero();
+
+    const fooTenpercent = foo(1.1);
+
+    fooTenpercent(100);
+
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion (non-blocking):** be less stupid, stupid.‏‏